### PR TITLE
Adds the ability to remove plating

### PR DESCRIPTION
**Pull request by Markolie**
**What does this PR do:**
This pull request adds the ability to remove plating using a welder, thus exposing the baseturf. This allows you to remove tiles you've placed on Lavaland so they become basalt again.

Right now this also allows you to remove plating on the station exposing the station to space, but I know when this was brought up two weeks or so ago that there were concerns with regards to griefing. If need be, I can easily make it so you can only remove turf that won't expose space (i.e. only Lavaland turfs).

**Changelog:**
:cl: Markolie
add: Plating can now be removed (exposing the baseturf) using a welder.
/:cl:

### CODE REVIEW

**Comment by datlo** [comment]
I would indeed recommend that plating cant be removed with just a welder on station. Engineering can get RCDs and start with one in CE office if they need to remove tiles for whatever reason.

**Comment by Xantholne** [comment]
I would say probably welder and then crowbar just to be safe if  it doesnt get a space check

**Comment by Lunaimon** [comment]
I think being able to cut holes in the station would be interesting, but definitely shouldn't be as simple as a single welder use. Maybe we could have reinforced plating with a complicated deconstruction process like reinforced walls & girders have?

**Comment by davidchan13** [comment]
Wonder if this would lead to plating melting away in plasma fires.

**Comment by Markolie** [comment]
It does not.

**Comment by datlo** [comment]
Plasma fires can already melt through station floor/plating. Needs to be EXTREMELY hot, though, a simple plasma cannister release wont do it alone usually.

**Comment by Citinited** [comment]
Before this merges, you will probably need to find a solution to the second part of #8815 as described in that issue (otherwise we'll get people randomly making holes in the station to space 😄 )

**Comment by KasparoVy** [comment]
RIPLEYs were the OGs, being able to remove plating before it was cool!

File: code/game/turfs/simulated/floor/plating.dm [hunk 1]
```diff
@@ -56,19 +56,38 @@
...
 				current_overlay = null
 				burnt = FALSE
 				broken = FALSE
 				update_icon()
+			else
+				to_chat(user, "<span class='notice'>You start removing [src].</span>")
+				playsound(src, welder.usesound, 100, 1)
+				if(do_after(user, 50 * welder.toolspeed, target = src) && welder && welder.isOn())
+					to_chat(user, "<span class='notice'>You remove [src].</span>")
+					new /obj/item/stack/tile/plasteel(get_turf(src))
```
[the remark this answers is not in the document]
Comment (Citinited) [reply code/game/turfs/simulated/floor/plating.dm]: That is the correct path for normal tiles made from metal - probably a relic from days gone by

[the remark this answers is not in the document]
Comment (Couls) [reply code/game/turfs/simulated/floor/plating.dm]: Huh you're right

**Pull request merged**